pipelines/controller/src/main/java/com/google/fhir/analytics/PipelineManager.java

```diff
@@ -211,15 +220,16 @@ private void initDwhStatus() throws ProfileException {
               prefix,
               DwhFiles.TIMESTAMP_PREFIX);
         }
-        if (lastDwh.isEmpty() || lastDwh.compareTo(path.getFilename()) < 0) {
+        if (lastDwh != null && (lastDwh.isEmpty() || lastDwh.compareTo(path.getFilename()) < 0)) {
           logger.debug("Found a more recent DWH {}", path.getFilename());
           lastDwh = path.getFilename();
         }
         // Do not consider if the DWH is not completely created earlier.
         if (!dwhFilesManager.isDwhComplete(path)) {
           continue;
         }
-        if (lastCompletedDwh.isEmpty() || lastCompletedDwh.compareTo(path.getFilename()) < 0) {
+        if (lastCompletedDwh != null
```

Comment:
ditto, let's keep `lastCompletedDwh` non-null. In general, for the fields/return-values that are in our control, it is best to try to avoid having `null` as an option if there is not too much extra overhead for it.